### PR TITLE
45 make rpc driving functions on client

### PR DESCRIPTION
**Pull request by Greve2001**
I have made a RobotController that simplifies making RPC calls.

### CODE REVIEW

File: remote/src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+public class Main {
+    public static void main(String[] args) throws InterruptedException {
+        RobotController controller = new RobotController();
+
+        controller.startController("192.168.1.12:50051");
```
Comment (MikaelFangel) [inline remote/src/main/java/Main.java:5]: I still think that the IP of the robot should be taken as a dynamic input, as originally intended.

```suggestion
        if (args.length < 1) {
            throw new MissingArgumentException("Please provide an IP and port number (e.g 192.168.0.97:50051)");
        }
        RobotController controller = new RobotController();

        controller.startController(args[0]);
        
```

File: remote/src/main/java/RobotController.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import proto.*;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+public class RobotController {
+    private ManagedChannel channel;
+    private MotorsGrpc.MotorsBlockingStub client;
+
+    private final int speed = 100;
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:13]: Why is this always 100 and a public field?

File: remote/src/main/java/RobotController.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * Used this method to start the controller before calling any other methods
+     * @param ip_port e.g. "192.168.1.12:50051"
+     */
+    public void startController(String ip_port) {
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:19]: If this always should be called before any other methods, then I think it should be a constructor and not a standalone method.

File: remote/src/main/java/RobotController.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+     * Use this method to stop the controller before ending the program.
+     * @throws InterruptedException if shutdown was interrupted
+     */
+    public void stopController() throws InterruptedException {
+        channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Makes the robot drive straight either forward or backwards for the distance given as argument
+     * @param distance Positive values for forward and negative for backwards
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:35]:
```suggestion
     * @param distance Positive values in cm for forward and negative for backwards
```

File: remote/src/main/java/RobotController.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+       ArrayList<MotorRequest> motorsRequest = createMotorRequests(Type.l, Port.A, Port.D);
+
+        DriveRequest driveRequest = DriveRequest.newBuilder()
+                .addAllMotors(motorsRequest)
+                .setDistance((float) distance)
+                .setSpeed(speed)
+                .build();
+
+        client.drive(driveRequest);
+
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:48]:
```suggestion
```

File: remote/src/main/java/RobotController.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        for (Port port : ports) {
+            motorRequests.add(MotorRequest.newBuilder()
+                    .setMotorPort(port)
+                    .setMotorType(motorType)
+                    .build());
+        }
+
+        return motorRequests;
+    }
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:80]: What should this be used for? And should it not also run the request as the other methods? (:

**Review by MikaelFangel — COMMENTED** [review]
Have some comments and questions before approval (:
 - But I like the idea of client methods.

File: remote/src/main/java/RobotController.java (hunk 6 above)
Comment (Greve2001) [reply remote/src/main/java/RobotController.java:80]: Its used by the other functions to create the individual request to each motor. It's just a helper function.

File: remote/src/main/java/RobotController.java (hunk 2 above)
Comment (Greve2001) [reply remote/src/main/java/RobotController.java:13]: Its the default speed. I will rename it in the next commit

File: remote/src/main/java/RobotController.java (hunk 3 above)
Comment (Greve2001) [reply remote/src/main/java/RobotController.java:19]: Great suggestion

File: remote/src/main/java/Main.java (hunk 1 above)
Comment (Greve2001) [reply remote/src/main/java/Main.java]: I implemented it in the newest commit

**Comment by Greve2001** [comment]
I have implemented your suggestions and explained some details you questioned.

**Review by MikaelFangel — APPROVED** [review]
Looks good then (:

**Pull request merged**